bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/NfcResource.java

```diff
@@ -157,15 +158,16 @@ public Response clearStore( final @ApiParam( name = "packageType", required = tr
             { @ApiResponse( code = 200, response = NotFoundCacheDTO.class, message = "The full not-found cache" ) } )
     @Produces( ApplicationContent.application_json )
     public Response getAll(
-                    final @ApiParam( name = "pageIndex", value = "page index, starts from 0" )
+                    final @ApiParam( name = "pageIndex", value = "page index" )
                     @QueryParam( "pageIndex" ) Integer pageIndex,
                     final @ApiParam( name = "pageSize", value = "page size" )
                     @QueryParam( "pageSize" ) Integer pageSize )
     {
         NotFoundCacheDTO dto;
-        if ( pageIndex != null && pageIndex >= 0 )
+        GenericPagination pagination = new NfcPagination(pageIndex, pageSize);
+        if ( pagination != null && pagination.allowPaging() )
```

Comment:
If the user does not submit the query param pageIndex, will it be null or 0? If it is null, the auto-unboxing will throw NPE. If it is zero 0, the allowPaging will return true which is not what we want ( we want getAll in such case). Can you double check this?

Reply:
@ruhan1 This will not have NPE problem, if user doesn't enter pageIndex, it should be null, and allowPaging has covered it, you may miss the latest commit: https://github.com/Commonjava/indy/pull/877/commits/076b8de8a6427bea0f746b8090d8d0b5855536de